File: apps/api-node/scripts/seed.ts
import { readFile } from "node:fs/promises";
import { resolve } from "node:path";
import { fileURLToPath } from "node:url";
import { buildApp } from "../src/app.js";
import { db } from "../src/db.js";

const fixtureDir = fileURLToPath(
  new URL("../../../shared/fixtures/", import.meta.url),
);
const app = await buildApp(db);

// Pass "wide" to load the extended fixture week instead of the brief's example data.
const datasets: Record<string, [string, string][]> = {
  demo: [
    ["LEDGER", "ledger.csv"],
    ["COUNTERPARTY", "counterparty.csv"],
  ],
  wide: [
    ["LEDGER", "ledger-wide.csv"],
    ["COUNTERPARTY", "counterparty-wide.csv"],
  ],
};
const dataset = process.argv[2] ?? "demo";
if (!datasets[dataset])
  throw new Error(`unknown dataset ${dataset}; choose demo or wide`);

for (const [source, filename] of datasets[dataset]) {
  const boundary = `atlas-seed-${source}`;
  const content = await readFile(resolve(fixtureDir, filename));
  const payload = Buffer.concat([
    Buffer.from(
      `--${boundary}\r\nContent-Disposition: form-data; name="file"; filename="${filename}"\r\nContent-Type: text/csv\r\n\r\n`,
    ),
    content,
    Buffer.from(`\r\n--${boundary}--\r\n`),
  ]);
  const response = await app.inject({
    method: "POST",
    url: `/api/files?source=${source}`,
    headers: { "content-type": `multipart/form-data; boundary=${boundary}` },
    payload,
  });
  if (response.statusCode >= 400) throw new Error(response.body);
  console.log(response.json());
}

await app.close();
await db.destroy();

File: apps/api-python/scripts/seed.py
from pathlib import Path
import sys
from fastapi.testclient import TestClient

ROOT = Path(__file__).resolve().parents[3]
sys.path.insert(0, str(ROOT / "apps/api-python"))
from app.main import app  # noqa: E402

# Pass "wide" to load the extended fixture week instead of the brief's example data.
DATASETS = {
    "demo": [("LEDGER", "ledger.csv"), ("COUNTERPARTY", "counterparty.csv")],
    "wide": [
        ("LEDGER", "ledger-wide.csv"),
        ("COUNTERPARTY", "counterparty-wide.csv"),
    ],
}
dataset = sys.argv[1] if len(sys.argv) > 1 else "demo"
if dataset not in DATASETS:
    raise SystemExit(f"unknown dataset {dataset}; choose demo or wide")

with TestClient(app) as client:
    for source, name in DATASETS[dataset]:
        path = ROOT / "shared/fixtures" / name
        response = client.post(
            f"/api/files?source={source}",
            files={"file": (name, path.read_bytes(), "text/csv")},
        )
        response.raise_for_status()
        print(response.json())
